Signup: enable hotjar based on flow prop

aneeshd16 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

## Proposed Changes

* This PR is a small refactor that enables hotjar via a `enableHotjar` prop on the flow config instead of hardcoding the flow names.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Since the `'ad-tracking'` config is only enabled in production, manually change the value to `true` and then re-run the Calypso build process.
https://github.com/Automattic/wp-calypso/blob/8abbe89d443c737d8ac47b9b5d243a5cfff096e8/config/development.json#L32
* Go to `/start/website-design-services?siteSlug=<siteSlug>` and confirm that the hotjar script has been loaded by inspecting the network log.


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Signup: enable hotjar based on flow prop

aneeshd16 commented:
Putting this on hold for now.

chriskmnds commented:
> Confirm that the hotjar script has been loaded by inspecting the network log for these flows:

@aneeshd16 how do we test this locally? look for a `hotjar.com` request?

aneeshd16 commented:
Hi, I updated the PR to enable hotjar only for the `site-content-collection` and `website-design-services` flows, but missed updating the testing instructions. I have now updated the test instructions. 🙂 

I can see the script loaded when I go to `/start/website-design-services?siteSlug=<site slug>`

<img width="1156" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/4ae06ccc-873b-4adb-a8ea-5cdd3d541345">

> perhaps I don't make the "region" cut on my end or something.

This is possible since there are numerous [checks](https://github.com/Automattic/wp-calypso/blob/a0c69020c98ace86b3a00d03affaf3884cbd3537/client/lib/analytics/tracker-buckets.ts#L143) that can prevent the script from loading.
